legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/IngestModeVisitors.java

```diff
@@ -37,7 +38,7 @@ private IngestModeVisitors()
         @Override
         public Boolean visitAppendOnly(AppendOnlyAbstract appendOnly)
         {
-            return appendOnly.filterExistingRecords();
+            return appendOnly.filterExistingRecords() || appendOnly.digestGenStrategy().accept(DIGEST_GEN_STRATEGY_DIGEST_REQUIRED);
```

Comment:
We can totally remove appendOnly.filterExistingRecords() check as that's covered in the validation

Reply:
Sure, changed!